text/2996-async-iterator.md

```diff
@@ -212,45 +213,45 @@ before they can be added.
-Stream` values without the need to monomorphize the functions that work
+AsyncIterator` values without the need to monomorphize the functions that work
 with them.
 
 Unfortunately, the use of poll does mean that it is harder to write
-stream implementations. The long-term fix for this, discussed in the [Future possiblilities][future-possibilities] section, is dedicated [generator syntax].
+async iterator implementations. The long-term fix for this, discussed in the [Future possiblilities][future-possibilities] section, is dedicated [generator syntax].
 
 # Rationale and alternatives
 [rationale-and-alternatives]: #rationale-and-alternatives
 
-## Where should stream live?
+## Where should async iterator live?
 
-As mentioned above, `core::stream` is analogous to `core::future`. But, do we want to find 
+As mentioned above, `core::async_iter` is analogous to `core::future`. But, do we want to find 
 some other naming scheme that can scale up to other future additions, such as io traits or channels?
 
 # Future possibilities
 [future-possibilities]: #future-possibilities
 
 ## Next method
 
-While users will be able to implement a `Stream` as defined in this RFC, they will not have a way to interact with it in the core library. As soon as we figure out a way to do it in an object safe manner, we should add a `next` method  either in the `Stream` trait or elsewhere.
+While users will be able to implement a `AsyncIterator` as defined in this RFC, they will not have a way to interact with it in the core library. As soon as we figure out a way to do it in an object safe manner, we should add a `next` method  either in the `AsyncIterator` trait or elsewhere.
 
-The `Iterator` trait includes a `next` method, which computes and returns the next item in the sequence. We should also implement a `next` method for `Stream`, similar to [the implementation in the futures-util crate](https://docs.rs/futures-util/0.3.5/src/futures_util/stream/stream/next.rs.html#10-12).
+The `Iterator` trait includes a `next` method, which computes and returns the next item in the sequence. We should also implement a `next` method for `AsyncIterator`, similar to [the implementation in the futures-util crate](https://docs.rs/futures-util/0.3.5/src/futures_util/stream/stream/next.rs.html#10-12).
 
 The core `poll_next` method is unergonomic; it does not let you iterate 
-over the items coming out of the stream. Therefore, we include a few minimal 
+over the items coming out of the async iterator. Therefore, we include a few minimal 
 convenience methods that are not dependent on any unstable features, such as `next`.
 
-As @yoshuawuyts states in their [pull request which adds `core::stream::Stream` to the standard library](https://github.com/rust-lang/rust/pull/79023):
+As @yoshuawuyts states in their [pull request which adds `core::async_iter::AsyncIterator` to the standard library](https://github.com/rust-lang/rust/pull/79023):
```

Comment:
Not sure replacing this makes sense, since that PR and discussion used the Stream naming.